asdf/fits_embed.py

```diff
@@ -235,22 +310,22 @@ def _update_asdf_extension(self, all_array_storage=None,
             all_array_compression=all_array_compression,
             auto_inline=auto_inline, pad_blocks=pad_blocks,
             include_block_index=False)
-        array = np.frombuffer(buff.getvalue(), np.uint8)
 
         try:
             asdf_extension = self._hdulist[ASDF_EXTENSION_NAME]
         except (KeyError, IndexError, AttributeError):
-            self._hdulist.append(fits.ImageHDU(array, name=ASDF_EXTENSION_NAME))
+            self._hdulist.append(self._create_hdu(buff, use_image_hdu))
         else:
```

Comment:
`array` is not defined here.

Reply:
Good catch. This was an oversight. I'm not sure it will be quite as simple as replacing the deleted line, but I'll add a unit test to exercise this case and go from there.